scripts/checkpoint_bots.py

```diff
@@ -237,17 +240,27 @@ def run_checkpoint_bot(
                     *fn_args,
                     nonce=nonce,
                 )
+                # Reset fail count on successful transaction
+                fail_count = 0
             except Exception as e:  # pylint: disable=broad-except
-                logging_str = (
-                    f"Pool {pool_name} for {checkpoint_time=}: checkpoint transaction failed, retrying. {repr(e)}.",
-                )
-                logging.warning(logging_str)
+                if fail_count < FAIL_COUNT_THRESHOLD:
+                    logging_str = "Checkpoint transaction failed."
+                    log_level = logging.WARNING
+                    term_logging_str = logging_str + f" {repr(e)}"
+                else:
+                    logging_str = f"Checkpoint transaction failed over {FAIL_COUNT_THRESHOLD} times."
+                    log_level = logging.CRITICAL
+                    term_logging_str = logging_str + f" {repr(e)}"
```

Comment:
looks like this line `term_logging_str = logging_str + f" {repr(e)}"` is only needed once after the conditional.